cmake/mex.cmake

```diff
@@ -225,21 +225,15 @@ function(mex_setup)
 
 
   # todo: handle C separately from CXX?
-  set (MEX_COMPILE_FLAGS "${MEX_INCLUDE} ${MEX_CXXFLAGS} ${MEX_DEFINES} ${MEX_MATLABMEX} ${MEX_CXX_ARGUMENTS}")
+  set (MEX_COMPILE_FLAGS "${MEX_CXXFLAGS} ${MEX_CXX_ARGUMENTS}")
   string(TOUPPER "${CMAKE_BUILD_TYPE}" _build_type)
   if (_build_type MATCHES DEBUG)
-    set(MEX_COMPILE_FLAGS "${MEX_CXXDEBUGFLAGS} ${MEX_COMPILE_FLAGS}")
     if (MSVC)
       string(REPLACE "/MD " "/MDd " MEX_COMPILE_FLAGS "${MEX_COMPILE_FLAGS}")
     endif()
-  elseif (_build_type MATCHES RELEASE)
-    set(MEX_COMPILE_FLAGS "${MEX_CXXOPTIMFLAGS} ${MEX_COMPILE_FLAGS}")
   endif()
 
-  if (${MEX_COMPILE_FLAGS} MATCHES "-ansi")
-    string(REPLACE "-ansi" "" MEX_COMPILE_FLAGS "${MEX_COMPILE_FLAGS}")
-    message(WARNING "Your MEX compiler flags contained '-ansi', but we've removed that flag for compatibility with C++11")
-  endif()
+  string(REPLACE "-ansi" "" MEX_COMPILE_FLAGS "${MEX_COMPILE_FLAGS}")
```

Comment:
Technically this does not affect compatibility with C++11 on supported platforms, though it would affect C11 compatibility.